lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -891,95 +893,44 @@ class ADContext {
   }
 
-  /// Finds a differentiation task on a function such that the task produces
-  /// adjoints for the specified indices.
-  DifferentiationTask *
-  lookUpDifferentiationTask(SILFunction *original,
-                            const SILAutoDiffIndices &indices) {
-    auto existing = enqueuedTaskIndices.find({original, indices});
-    if (existing == enqueuedTaskIndices.end())
-      return nullptr;
-    return differentiationTasks[existing->getSecond()].get();
-  }
-
-  /// Finds a differentiation task on a function such that the task produces
-  /// adjoints for the specified indices or, if such a task is not present,
-  /// for the task with the least number of parameters that is a superset of
-  /// the parameter indices in `indices`, and which corresponds to a
-  /// primitive adjoint function.
-  DifferentiationTask *
-  lookUpMinimalDifferentiationTask(SILFunction *original,
-                                   const SILAutoDiffIndices &indices) {
-    auto *superset = AutoDiffIndexSubset::getDefault(
-        getASTContext(),
-        original->getLoweredFunctionType()->getNumParameters(), false);
-    auto *indexSet = indices.parameters;
-    if (auto *existingTask = lookUpDifferentiationTask(original, indices))
-      return existingTask;
-    for (auto *rda : original->getDifferentiableAttrs()) {
-      auto *rdaIndexSet = rda->getIndices().parameters;
-      // If all indices in `indexSet` are in `rdaIndexSet`, and it has fewer
-      // indices than our current candidate and a primitive VJP, then `rda` is
-      // our new candidate.
-      //
-      // NOTE: `rda` may come from a un-partial-applied function, it may have
-      // more parameters than the desired indices. We expect this logic to go
-      // away when we support `@differentiable` partial apply.
-      if (rdaIndexSet->isSupersetOf(
-              indexSet->extendingCapacity(getASTContext(),
-                                          rdaIndexSet->getCapacity())) &&
-          // fewer parameters than before
-          (superset->isEmpty() ||
-           rdaIndexSet->getNumIndices() < superset->getNumIndices()))
-        superset = rda->getIndices().parameters;
-    }
-    auto existing =
-        enqueuedTaskIndices.find({original, {indices.source, superset}});
-    if (existing == enqueuedTaskIndices.end())
-      return nullptr;
-    return differentiationTasks[existing->getSecond()].get();
-  }
-
-  /// Tries to promote origFnOperand which is of function type to an autodiff
-  /// function.
-  SILValue promoteToDifferentiableFunction(
-      SILBuilder &builder, SILLocation loc, SILValue origFnOperand,
-      AutoDiffIndexSubset *parameterIndices,
-      unsigned differentiationOrder, DifferentiationInvoker invoker);
+private:
+  /// Canonicalizes the given `autodiff_function` instruction, populating
+  /// missing associated functions, and returns
+  SILValue getCanonicalizedAutoDiffFunctionInst(
```

Comment:
We discussed in person. The semantics of the method are a bit complicated.
We agreed to use the original name of this method, `promoteToDifferentiableFunction`, since the method always returns a `SILValue` with `@differentiable` function type.